lsp/state/data/mips_instruction.cpp

```diff
@@ -0,0 +1,531 @@
+    {"vu", "VLQD", "Load with Pre-Decrement"},
+    {"vu", "VLQI", "Load with Post-Increment"},
+    {"vu", "VMADD", "Product Sum"},
+    {"vu", "VMADDi", "Product Sum with I Register"},
+    {"vu", "VMADDq", "Product Sum with Q Register"},
+    {"vu", "VMADDbc", "Broadcast Product Sum"},
+    {"vu", "VMADDA", "Product Sum to Accumulator"},
+    {"vu", "VMADDAi", "Product Sum with I Register, to Accumulator"},
+    {"vu", "VMADDAq", "Product Sum with Q Register, to Accumulator"},
+    {"vu", "VMADDAbc", "Broadcast Product Sum to Accumulator"},
+    {"vu", "VMAX", "Maximum Value"},
+    {"vu", "VMAXi", "Maximum Value"},
+    {"vu", "VMAXbc", "Maximum Value"},
+    {"vu", "VMFIR", "Transfer from Integer Register to Floating-Point Register"},
+    {"vu", "VMINI", "Minimum Value"},
+    {"vu", "VMINIi", "Minimum Value"},
+    {"vu", "VMINIbc", "Minimum Value"},
+    {"vu", "VMOVE", "Transfer between Floating-Point Registers"},
+    {"vu", "VMR32", "Vector Rotate"},
+    {"vu", "VMSUB", "Multiply and Subtract"},
+    {"vu", "VMSUBi", "Multiply and Subtract with I Register"},
+    {"vu", "VMSUBq", "Multiply and Subtract Q Register"},
+    {"vu", "VMSUBbc", "Broadcast Multiply and Subtract"},
+    {"vu", "VMSUBA", "Multiply and Subtract to Accumulator"},
+    {"vu", "VMSUBAi", "Multiply and Subtract with I Register, to Accumulator"},
+    {"vu", "VMSUBAq", "Multiply and Subtract with Q Register, to Accumulator"},
+    {"vu", "VMSUBAbc", "Broadcast Multiply and Subtract to Accumulator"},
+    {"vu", "VMTIR", "Transfer from Floating-Point Register to Integer Register"},
+    {"vu", "VMUL", "Multiply"},
+    {"vu", "VMULi", "Multiply by I Register"},
+    {"vu", "VMULq", "Multiply by Q Register"},
+    {"vu", "VMULbc", "Broadcast Multiply"},
+    {"vu", "VMULA", "Multiply to Accumulator"},
+    {"vu", "VMULAi", "Multiply by I Register to Accumulator"},
+    {"vu", "VMULAq", "Multiply by Q Register to Accumulator"},
+    {"vu", "VMULAbc", "Broadcast Multiply to Accumulator"},
+    {"vu", "VNOP", "No Operation"},
+    {"vu", "VOPMULA", "Vector Outer Product"},
+    {"vu", "VOPMSUB", "Vector Outer Product"},
+    {"vu", "VRGET", "Get Random Numbers"},
+    {"vu", "VRINIT", "Random Number Initial Set"},
+    {"vu", "VRNEXT", "New Random Numbers"},
+    {"vu", "VRSQRT", "Square Root Division"},
+    {"vu", "VRXOR", "Random Number Set"},
+    {"vu", "VSQD", "Store with Pre-Decrement"},
+    {"vu", "VSQI", "Store with Post-Increment"},
+    {"vu", "VSQRT", "Square Root"},
+    {"vu", "VSUB", "Subtract"},
+    {"vu", "VSUBi", "Subtract I Register"},
+    {"vu", "VSUBq", "Subtract Q Register"},
+    {"vu", "VSUBbc", "Broadcast Subtract"},
+    {"vu", "VSUBA", "Subtract to Accumulator"},
+    {"vu", "VSUBAi", "Subtract I Register to Accumulator"},
+    {"vu", "VSUBAq", "Subtract Q Register to Accumulator"},
+    {"vu", "VSUBAbc", "Broadcast Subtract to Accumulator"},
+    {"vu", "VWAITQ", "Q Register Synchronize"},
```

Comment:
It might be possible to reuse the `OpcodeInfo.cpp` stuff, if you eventually want to have more information about the the number/kind of arguments to each op.

Reply:
Hmm yeah that could be useful, id need a way to go from a mnemonic -> `InstructionKind` though i think?

Reply:
yeah it would require some work and probably isn't worth it unless you want to do more stuff with the assembly ops.